internal/app/init/pkg/network/setup.go

```diff
@@ -0,0 +1,104 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package network
+
+import (
+	"log"
+
+	"github.com/pkg/errors"
+	"github.com/talos-systems/talos/pkg/userdata"
+	"github.com/vishvananda/netlink"
+)
+
+// InitNetwork handles the initial network setup.
+// This includes doing ifup for lo and eth0 and
+// the initial ip addressing via dhcp for eth0
+func InitNetwork() (err error) {
```

Comment:
I'm not that good at naming stuff, but I usually read that Go doesn't like when package name is repeated in the identifier inside the package.

So whether it is `network.InitNetwork()` or `network.Init()` :)

Reply:
I was concerned of the collision with go's use of  `init()`. I know the cases are different ( init vs Init ) but thought that'd lead to unnecessary confusion. 
Naming stinks.